File: src/net/nuggetmc/mw/command/getItemCommand.java
package net.nuggetmc.mw.command;

import net.nuggetmc.mw.MegaWalls;
import net.nuggetmc.mw.mwclass.MWClass;
import net.nuggetmc.mw.mwclass.MWClassManager;
import net.nuggetmc.mw.mwclass.items.MWPotions;
import net.nuggetmc.mw.special.SpecialItemUtils;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.PlayerInventory;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;
import java.util.Objects;

public class getItemCommand implements CommandExecutor, TabCompleter {
    private final List<String> groupnames = new ArrayList<>();
    private final MWClassManager classManager = MegaWalls.getInstance().getClassManager();
    private final SpecialItemUtils si;

    public getItemCommand() {
        this.groupnames.add("healpot");
        this.groupnames.add("speedpot");
        this.groupnames.add("squpot");
        this.groupnames.add("golempot");
        this.groupnames.add("milk");
        this.si = MegaWalls.getInstance().getSpecialItemUtils();
    }



    public final MWClassManager getClassManager() {
        return this.classManager;
    }



    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if (!(sender instanceof Player)) return true;
        Player player= ((Player) sender);
        if (!MegaWalls.getInstance().getCombatManager().isInCombat(player)) {
            return true;
        }
        if (args.length == 2 || args.length == 1) {
            int amount;
            if (args.length == 2) {
                try {
                    amount = Integer.parseInt(args[1]);
                } catch (Exception e) {
                    sender.sendMessage("invalid syntax!Correct usage: /mwitem item count");
                    return true;
                }
            } else {
                amount = 1;
            }
            ItemStack itemStack;
            switch (args[0]){
                case "healpot":
                    itemStack=MWPotions.createHealPotions(
                            classManager.get(player).getName(),
                        classManager.get(player).getColor(),
                        amount,
                        10
                );
                     break;
                case "speedpot":
                    itemStack=MWPotions.createSpeedPotions(
                            classManager.get(player).getName(),
                        classManager.get(player).getColor(),
                        amount
                );
                    break;
                case "squpot":
                    itemStack=si.getSquidPot(amount);
                    break;
                case "golempot":
                    itemStack=si.getGolemPot(amount);
                    break;
                case "milk":
                    itemStack=si.getCowBucket(amount);
                    break;
                default:
                    itemStack=null;
            }
            if (itemStack == null) {
                sender.sendMessage("invalid item!Allowed items: "+groupnames);
            } else {
                player.getInventory().addItem(itemStack);
            }
        } else {
            sender.sendMessage("invalid syntax!Correct usage: /mwitem item count");
            return true;
        }
        return true;
    }

    public List<String> onTabComplete(CommandSender sender, Command cmd, String label,String[] args) {
        if (args.length != 1) {
            return null;
        } else {
            String arg = args[0];
            return !this.isEmptyTab(arg) ? this.autofill(this.groupnames, arg) : this.groupnames;
        }
    }

    private List<String> autofill(List<String> groupnames, String input) {
        List<String> list = new ArrayList<>();

        for (String entry : groupnames) {
            if (entry.length() >= input.length()) {
                String name = entry.substring(0, input.length());
                if (Objects.equals(input, name)) {
                    list.add(entry);
                }
            }
        }

        return list.isEmpty() ? groupnames : list;
    }

    private boolean isEmptyTab(String s) {
        return s == null || Objects.equals(s, " ") || ((CharSequence)s).length() == 0;
    }

}

File: src/net/nuggetmc/mw/command/GetItemCommand.kt
package net.nuggetmc.mw.command

import net.nuggetmc.mw.MegaWalls
import net.nuggetmc.mw.mwclass.items.MWPotions
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.entity.Player

class GetItemCommand : CommandExecutor, TabCompleter {
    val groupnames: MutableList<String> = ArrayList()
    val classManager = MegaWalls.getInstance().classManager

    init {
        groupnames.add("healpot")
        groupnames.add("speedpot")
        groupnames.add("squpot")
        groupnames.add("golempot")
        groupnames.add("milk")
        groupnames.add("aotv")
        groupnames.add("hype")
        groupnames.add("term")
    }

    val si = MegaWalls.getInstance().specialItemUtils
    override fun onCommand(sender: CommandSender?, cmd: Command?, label: String?, args: Array<String?>?): Boolean {
        if (!(sender is Player)) return true
        val player = sender
        if (!MegaWalls.getInstance().combatManager.isInCombat(player)) {
            return true
        }
        if (args?.size == 2 || args?.size == 1) {
            val amount: Int
            if (args.size == 2) {
                try {
                    amount = Integer.parseInt(args[1])
                } catch (e: Exception) {
                    sender.sendMessage("invalid syntax!Correct usage: /mwitem item count")
                    return true
                }
            } else {
                amount = 1
            }
            var itemStack = when (args[0]) {
                "healpot" -> MWPotions.createHealPotions(
                    classManager?.get(player)?.name,
                    classManager.get(player).color,
                    amount,
                    10
                )

                "speedpot" -> MWPotions.createSpeedPotions(
                    classManager?.get(player)?.name,
                    classManager.get(player).color,
                    amount
                )

                "squpot" -> si.getSquidPot(amount)
                "golempot" -> si.getGolemPot(amount)
                "milk" -> si.getCowBucket(amount)
                "aotv" -> si.aotv
                "hype" -> si.hyperion
                "term" -> si.term
                else -> {
                    null
                }
            }
            if (itemStack == null) {
                sender.sendMessage("invalid item!Allowed items:$groupnames")
            } else {
                player.inventory.addItem(itemStack)
            }
        } else {
            sender.sendMessage("invalid syntax!Correct usage: /mwitem item count")
            return true
        }
        return true
    }

    override fun onTabComplete(
        sender: CommandSender?,
        cmd: Command?,
        label: String?,
        args: Array<String>
    ): List<String>? {
        if (args.size != 1) return null

        val arg = args[0]
        return if (!isEmptyTab(arg)) {
            autofill(groupnames, arg)
        } else groupnames
    }


    private fun autofill(groupnames: List<String>, input: String): List<String> {
        val list: MutableList<String> = ArrayList()
        for (entry in groupnames) {
            if (entry.length >= input.length) {
                if (input.equals(entry.substring(0, input.length), ignoreCase = true)) {
                    list.add(entry)
                }
            }
        }
        return if (list.isEmpty()) {
            groupnames
        } else list
    }

    private fun isEmptyTab(s: String?): Boolean {
        return s == null || s == " " || s.isEmpty()
    }
}